### PR TITLE
Tidy GPS pre-arm checks

### PR DESCRIPTION
**Pull request by peterbarker**
Will allow backends to have their own pre-arm checks.

Rename pre-arm checks to be consistent with our libraries.  Note that the pre-arm check doesn't actually check the backends are healthy, despite the previous name...

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS_DroneCAN.cpp [hunk 1]
```diff
@@ -225,8 +225,24 @@ AP_GPS_Backend* AP_GPS_DroneCAN::probe(AP_GPS &_gps, AP_GPS::GPS_State &_state)
     return backend;
 }
 
-bool AP_GPS_DroneCAN::backends_healthy(char failure_msg[], uint16_t failure_msg_len)
+bool AP_GPS_DroneCAN::pre_arm_checks(char failure_msg[], uint16_t failure_msg_len)
 {
+    // only do these checks if there is at least one DroneCAN GPS:
+    bool found_uavan_type = false;
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_DroneCAN.cpp:231]: uavan -> DroneCAN

File: libraries/AP_GPS/AP_GPS_DroneCAN.cpp [hunk 2]
```diff
@@ -225,8 +225,24 @@ AP_GPS_Backend* AP_GPS_DroneCAN::probe(AP_GPS &_gps, AP_GPS::GPS_State &_state)
...
+        }
+        found_uavan_type = true;
+   }
+    if (!found_uavan_type) {
+        return true;
+    }
+
+    // lint parameters and detected node IDs:
     for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
         const auto &params_i = AP::gps().params[i];
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_DroneCAN.cpp:247]: maybe check for DroneCAN inside loop, and remove above?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_GPS/AP_GPS_DroneCAN.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS_DroneCAN.cpp]: This line's now been removed

File: libraries/AP_GPS/AP_GPS_DroneCAN.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS_DroneCAN.cpp:232]: Done

**Review by tridge — APPROVED** [review]

**Pull request merged**